src/main/java/baritone/pathing/precompute/PrecomputedData.java

```diff
@@ -22,46 +22,47 @@
 import net.minecraft.world.level.block.Block;
 import net.minecraft.world.level.block.state.BlockState;
 
-import static baritone.pathing.precompute.Ternary.MAYBE;
-import static baritone.pathing.precompute.Ternary.YES;
-
 public class PrecomputedData {
 
-    private final int[] data = new int[Block.BLOCK_STATE_REGISTRY.size()];
+    private final byte[] data = new byte[Block.BLOCK_STATE_REGISTRY.size()];
+
+    /**
+     * byte layout
+     *
+     *          0              1             2              3              4              5              6             7
+     *          |              |             |              |              |              |              |             |
+     *      unused         canWalkOn       maybe       canWalkThrough    maybe        fullyPassable    maybe       completed
+     */
+
+    private static final byte COMPLETED_MASK = (byte) 0x01;
+    private static final byte FULLY_PASSABLE_MAYBE_MASK = (byte) 0x02;
+    private static final byte FULLY_PASSABLE_MASK = (byte) 0x04;
+    private static final byte CAN_WALK_THROUGH_MAYBE_MASK = (byte) 0x08;
+    private static final byte CAN_WALK_THROUGH_MASK = (byte) 0x10;
+    private static final byte CAN_WALK_ON_MAYBE_MASK = (byte) 0x20;
+    private static final byte CAN_WALK_ON_MASK = (byte) 0x40;
 
-    private static final int COMPLETED_MASK = 1 << 0;
-    private static final int CAN_WALK_ON_MASK = 1 << 1;
-    private static final int CAN_WALK_ON_SPECIAL_MASK = 1 << 2;
-    private static final int CAN_WALK_THROUGH_MASK = 1 << 3;
-    private static final int CAN_WALK_THROUGH_SPECIAL_MASK = 1 << 4;
-    private static final int FULLY_PASSABLE_MASK = 1 << 5;
-    private static final int FULLY_PASSABLE_SPECIAL_MASK = 1 << 6;
+    private static final byte zero = (byte) 0;
```

Comment:
huh??

Reply:
is this to prevent implicit upcasting to int on both sides when you would do `myByte == 0`?